src/docker/manifest.nim

```diff
@@ -351,3 +320,94 @@ proc fetchImageManifest*(name: DockerImage,
       $platform & " != " & "" & $manifest.platform
     )
   return manifest
+
+proc findSibling(self: DockerManifest, reference = "attestation-manifest"): DockerManifest =
+  if self.kind != DockerManifestType.image:
+    raise newException(ValueError, "Can only lookup sibling for image manifest")
+  if self.list == nil:
+    raise newException(ValueError, "Need reference to list manifest to lookup sibling")
+  for i in self.list.manifests:
+    if i.annotations == nil:
+      continue
+    if (
+      i.annotations{"vnd.docker.reference.type"}.getStr().toLower() == reference.toLower() and
+      i.annotations{"vnd.docker.reference.digest"}.getStr().toLower() == self.nameRef.imageRef.toLower()
+    ):
+      return i.fetch()
+  raise newException(KeyError, "Could not find sibling image of reference type: " & reference)
+
+proc findInTotoLayer(self: DockerManifest, predicate: string): DockerManifest =
+  if self.kind != DockerManifestType.image:
+    raise newException(ValueError, "Can only lookup layers in image manifest")
+  for l in self.layers:
+    if l.annotations == nil:
+      continue
+    if l.annotations{"in-toto.io/predicate-type"}.getStr().toLower().startsWith(predicate.toLower()):
```

Comment:
`333,334,345`: util function for case-insensitive string comparisons?

Reply:
🤷 . I like using explicit `==` for comparisons as most editors highlight `==` differently vs utility will have to be something like `a.foo(b)` which wont use `==` unless we add string wrapper around it like `Caseless(a) == b`, it can convert inside `==` implicitly but that can be hard to follow as well

as its all one-line, prolly no big deal but no strong feeling either way